player/models.py

```diff
@@ -35,20 +40,27 @@ def create(username, password, birth_date, favorite_ball_game):
         player.save()
         return player
 
-    def validate_and_save(self, birth_date, favorite_ball_game):
+    def validate_and_save(self, birth_date, favorite_ball_game, profile_picture=None):
         errors = []
         try:
             datetime.datetime.strptime(birth_date, '%Y-%m-%d').date()
             self.birth_date = birth_date
         except ValueError:
-            errors.append("Invalid birth date format. Please use the format YYYY-MM-DD.")
+            errors.append("Invalid birth date format, Please use the format YYYY-MM-DD.")
 
         ball_games_names = [choice[0] for choice in BallGame.choices]
         if favorite_ball_game not in ball_games_names:
-            errors.append("Invalid favorite ball game. Please select a valid option.")
+            errors.append("Invalid favorite ball game, Please select a valid option.")
         else:
             self.favorite_ball_game = favorite_ball_game
 
+        if profile_picture:
+            file_extension = profile_picture.name.split('.')[-1].lower()
+            if file_extension not in ['png', 'jpeg', 'jpg']:
```

Comment:
See if you can use `endswith()` instead of implementing this manipulation yourself:
```suggestion
            if not profile_picture.name.endswith(('.png', '.jpeg', '.jpg')):
```

Reply:
fixed

Reply:
My suggestion dropped the call to `lower` which was smart. Maybe you can add it. i.e. `.name.lower.endswith(...`